includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -1338,6 +1338,10 @@ private function is_valid_layout( $tag_spec, $node ) {
 			return true;
 		}
 
+		if ( $this->is_inside_mustache_template( $node ) ) {
+			return true;
```

Comment:
Maybe this goes too far in not validating the layout at all if it's inside a `<template type="amp-mustache">` or `<script type="text/plain" template="amp-mustache>"`

Reply:
Let's use the AMP validator as the guide:

![image](https://user-images.githubusercontent.com/134745/79275148-86af1900-7e5a-11ea-925b-85518dd391e4.png)

If any of the layout attributes (`layout`, `width`, `height`, `heights`, or `sizes`) have a Mustache template variable in them, then the validation is skipped.

This is the logic that we need to replicate: https://github.com/ampproject/amphtml/blob/c083d2c6120a251dcc9b0beb33c0336c7d3ca5a8/validator/engine/validator.js#L4038-L4054

Example AMP page with samples:

```html
<!doctype html>
<html ⚡>
<head>
	<meta charset="utf-8">
	<link rel="canonical" href="self.html" />
	<meta name="viewport" content="width=device-width,minimum-scale=1">
	<style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
	<script async src="https://cdn.ampproject.org/v0.js"></script>
	<script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.2.js"></script>
</head>
<body>
	<template type="amp-mustache"><div><div><div><amp-img src="/img1.png" layout="bad" width="bad" height="bad"></amp-img></div></div></div></template>
	<template type="amp-mustache"><div><div><div><amp-img src="/img1.png" layout="{{layout}}" width="bad" height="bad"></amp-img></div></div></div></template>
	<template type="amp-mustache"><div><div><div><amp-img src="/img1.png" layout="bad" width="{{width}}" height="bad"></amp-img></div></div></div></template>
	<template type="amp-mustache"><div><div><div><amp-img src="/img1.png" layout="bad" width="bad" height="{{height}}"></amp-img></div></div></div></template>
	<template type="amp-mustache"><div><div><div><amp-img src="/img1.png" layout="bad" width="bad" height="bad" heights="{{height}}"></amp-img></div></div></div></template>
	<template type="amp-mustache"><div><div><div><amp-img src="/img1.png" layout="bad" width="bad" height="bad" heights="bad" sizes="{{sizes}}"></amp-img></div></div></div></template>
</body>
</html>
```

Reply:
Sounds good, I'll make a commit for that if you'd like.

Reply:
Augment the test cases with the samples above.

Reply:
Sure, https://github.com/ampproject/amp-wp/pull/4583/commits/3d994a91e4f580728519e04cdf0f94d552bbfaaa applies this suggestion. 

I'll add some more test cases.